sky/serve/serve_state.py

```diff
@@ -175,6 +178,8 @@ def from_replica_statuses(
         if sum(status2num[status]
                for status in ReplicaStatus.failed_statuses()) > 0:
             return cls.FAILED
+        if len(replica_statuses) == 0:
+            return cls.NO_REPLICA
```

Comment:
What if there are some failed replica in the system?

Reply:
Changed it to 
```
if (len(replica_statuses) - status2num[ReplicaStatus.FAILED] -
                status2num[ReplicaStatus.FAILED_CLEANUP] == 0):
```